packages/react-i18n/docs/react_i18n_schema.md

```diff
@@ -0,0 +1,284 @@
+
+**Example React I18n JSON file**
+
+```jsonc
+{
+  "hello": "My name is {name}",
+  "parent": {
+    "child_1": "I am the first child of `parent`!",
+    "child_2": {
+      "grandchild_1": "I am the first grandchild of `parent`!",
+      "grandchild_2": "I am the second grandchild of `parent`!",
+    }
+  }
+}
+```
+
+# Pluralization
+
+Pluralization is handled by providing the pluarlization keys required by the language as leaf KVPs, creating a "pluralization context".
+
+**Example English file with pluralization keys:**
+```jsonc
+{
+  "cars": {  // All the children of `cars` are in a "pluralization context"
+    "one": "I have {count} car",
+    "other": "I have {count} cars"
+  }
+}
+```
+
+Different languages require different sets of pluralization keys. For example, the Polish (`pl`) has 4 required pluralization keys: `one`, `few`, `many`, and `other`. The Polish version of the previous example would be:
+
+**Example Polish file with pluralization keys:**
+```jsonc
+{
+  "cars": {
+    "one": "Mam {count} samochód",
+    "few": "Mam {count} samochody",
+    "many": "Mam {count} samochodów",
+    "other": "Mam {count} samochodu",
+  }
+}
+```
+
+## Reserved pluralization keys
+
+In order to avoid collisions, the following keys are reserved for use as pluralization keys, and must not be used as keys of leaf KVPs outside of the pluralization context:
+
+* `few`
+* `many`
+* `one`
+* `other`
+* `two`
+* `zero`
+
+These keys are derived from the names given to the pluralization rules of all languages, as defined in the [Unicode Consortium's Common Locale Data Repository (CLDR)](https://github.com/unicode-org/cldr).
+
+This list is current as of verison `37` of the CLDR. In the unlikely event that new pluralization rules names are added, they will be added to this reserved list in a future version of this specification.
+
+**Example INVALID English file:**
+```jsonc
+{
+  "foo": "bar",
+  "other": "Other" // INVALID: `other` is a keyword reserved for use in pluralization contexts!
+}
+```
+
+## Invalid pluralization keys
+
+Further, pluralization keys that are not required by the language must not be used, even within a pluralization context.
+
+For example, even though `zero` is a required pluralization key for some languages (eg. Arabic), it is not a required pluralization key in English. Therefore, the `zero` key should not be present in an English file alongside the pluralization KVPs required by English.
+
+**Example INVALID English file:**
+```jsonc
+{
+  "cars": {
+    "zero": "I have no cars", // INVALID: `zero` is not a required pluralization key for English!
+    "one": "I have {count} car",
+    "other": "I have {count} cars"
+  }
+}
+```
+
+## Detecting pluralization contexts
+
+These constraints allow for an algorithm for identifying pluralization contexts:
+
+If a node has child leaf KVPs for each of the pluralization keys required by the language, then its children are within a pluralization context.
```

Comment:
True, but could this be strengthened to simply "If a node has child leaf KVPs for any of the pluralization keys, then ..."? Since pluralization keys outside this context are invalid?

Reply:
You are right. Fixed.